# Week 5 - Final Assignment

## Description of the problem and a discussion of the background.

A friend of mine is a good cook and moving from another city to Chicago with a plan to open a restuarant. He asked my help to find a best community that fits his needs. 
His requirements are, The place and surroundings should have sizable middle age people and good employment rate. Also since he is going to open a Mexican restuarant there should be a sizable Latino and Hispanic community in the neighbourhood but not with many mexican restuarants around. Also his resturant price range is going to be Moderate to High so the community income should be decent to afford.

## Datasets

To Solve the problem, I need the following datasets:
    
    1) As the restuarant is going to carry moderate to high price range dishes , I need the Communities in Chicago, their employment rates and per capita income

    2) As the need is to open in a community with decent employment rates and sizeable Latino community and middle age groups, I need a dataset with demographic information with age.

    3) I need to query the foursquare API to address the final need, make sure there are not many Mexican resturants arounds and their price ranges to get an idea of the community


After careful research, Here are a few datasets I have found that helps me find the best place that fits his needs:
    
    Census data to help with first dataset need above:
    
        https://datahub.cmap.illinois.gov/dataset/2010-census-data-summarized-to-chicago-community-areas
    
    Census data to help with first dataset need above:
    
        https://data.cityofchicago.org/Health-Human-Services/Census-Data-Selected-socioeconomic-indicators-in-C/kn9c-c2s2

    Finally Foursquare API for price ranges and to find other Mexican restuarants with similar price ranges.

### Plan of execution:
Step 1: Import the two datasets as CSV, remove the attributes that are not needed, merge some attributes as per need and changes cases/types

Step 2: Create three Ranking columns 
    
        1) Rank on the Hispanic and latino population counts 
        
        2) Rank on Income of the community 
        
        3) Rank on the age groups

Step 3: Connect to foursquare and create a list of all the mexican restuarants and their price tier by community

Step 3: Combine the foursquare data and the community data and find the top two communities that really fit the requirements


### Step 1: 
### Import the two datasets as CSV, remove the attributes that are not needed, merge some attributes as per need and changes cases/types


In [1]:

import random # library for random number generation
import numpy as np # library for vectorized computation
import pandas as pd # library to process data as dataframes

from sklearn.cluster import KMeans 

print('Libraries imported.')


Libraries imported.


In [2]:
### In this step we are importing the CSV which contains the demographic information about each community
Chicago_Neigh_Demo = pd.read_csv('Demographics_Chicago_2008_2012.csv')
Chicago_Neigh_Demo

Community Area Number COMMUNITY AREA NAME  PERCENT OF HOUSING CROWDED  \
0                    14.0         Albany Park                        11.3   
1                    57.0      Archer Heights                         8.5   
2                    34.0       Armour Square                         5.7   
3                    70.0             Ashburn                         4.0   
4                    71.0      Auburn Gresham                         4.0   
..                    ...                 ...                         ...   
73                   65.0           West Lawn                         5.8   
74                   53.0        West Pullman                         3.3   
75                    2.0          West Ridge                         7.8   
76                   24.0           West Town                         2.3   
77                   42.0            Woodlawn                         2.9   

    PERCENT HOUSEHOLDS BELOW POVERTY  PERCENT AGED 16+ UNEMPLOYED  \
0                               19.2                         10.0   
1                               14.1                         16.5   
2                               40.1                         16.7   
3                               10.4                         11.7   
4                               27.6                         28.3   
..                               ...                          ...   
73                              14.9                          9.6   
74                              25.9                         19.4   
75                              17.2                          8.8   
76                              14.7                          6.6   
77                              30.7                         23.4   

    PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA  \
0                                           32.9   
1                                           35.9   
2                                           34.5   
3                                           17.7   
4                                           18.5   
..                                           ...   
73                                          33.6   
74                                          20.5   
75                                          20.8   
76                                          12.9   
77                                          16.5   

    PERCENT AGED UNDER 18 OR OVER 64  PER CAPITA INCOME   HARDSHIP INDEX  
0                               32.0               21323            53.0  
1                               39.2               16134            67.0  
2                               38.3               16148            82.0  
3                               36.9               23482            37.0  
4                               41.9               15528            74.0  
..                               ...                 ...             ...  
73                              39.6               16907            56.0  
74                              42.1               16563            62.0  
75                              38.5               23040            46.0  
76                              21.7               43198            10.0  
77                              36.1               18672            58.0  

[78 rows x 9 columns]

In [3]:
!pip install xlrd
### In this step we are importing an excel which contains the Community information and Hispanic and Latino community details in particular
Chicago_Neigh_Data = pd.read_excel (r'Chicago_Neighborhoods.xlsx') 
Chicago_Neigh = pd.DataFrame(Chicago_Neigh_Data, columns = ['Geog','GeogKey','Total Population','Not Hispanic or Latino but White alone','Not Hispanic or Latino but Black or African American alone','Not Hispanic or Latino but American Indian and Alaska Native alone','Not Hispanic or Latino but Asian alone','Not Hispanic or Latino but Native Hawaiian and Other Pacific Islander alone','Not Hispanic or Latino but Some Other Race alone','Not Hispanic or Latino but Two or More Races','Hispanic or Latino','Male:  to 5 yrs','Male: 5 to 9 yrs','Male: 10 to 14 yrs','Male: 15 to 17 yrs','Male: 18 and 19 yrs','Male: 20 yrs','Male: 21 yrs','Male: 22 to 24 yrs','Male: 25 to 29 yrs','Male: 30 to 34 yrs','Male: 35 to 39 yrs','Male: 40 to 44 yrs','Male: 45 to 49 yrs','Male: 50 to 54 yrs','Male: 55 to 59 yrs','Male: 60 to 61 yrs','Male: 62 to 64 yrs','Male: 65 to 66 yrs','Male: 67 to 69 yrs','Male: 70 to 74 yrs','Male: 75 to 79 yrs','Male: 80 to 84 yrs','Male: 85 yrs and over','FeMale: Under 5 yrs','FeMale: 5 to 9 yrs','FeMale: 10 to 14 yrs','FeMale: 15 to 17 yrs','FeMale: 18 to 19 yrs','FeMale: 20 yrs','FeMale: 21 yrs','FeMale: 22 to 24 yrs','FeMale: 25 to 29 yrs','FeMale: 30 to 34 yrs','FeMale: 35 to 39 yrs','FeMale: 40 to 44 yrs','FeMale: 45 to 49 yrs','FeMale: 50 to 54 yrs','FeMale: 55 to 59 yrs','FeMale: 60 to 61 yrs','FeMale: 62 to 64 yrs','FeMale: 65 to 66 yrs','FeMale: 67 to 69 yrs','FeMale: 70 to 74 yrs','FeMale: 75 to 79 yrs','FeMale: 80 to 84 yrs','FeMale: 85 yrs and over','Median Age','Total Households'])
Chicago_Neigh.head(5)

     |████████████████████████████████| 112kB 3.1MB/s eta 0:00:01


Geog  GeogKey  Total Population  \
0     Rogers Park        1             54991   
1      West Ridge        2             71942   
2          Uptown        3             56362   
3  Lincoln Square        4             39493   
4    North Center        5             31867   

   Not Hispanic or Latino but White alone  \
0                                   21618   
1                                   30706   
2                                   29098   
3                                   24916   
4                                   24634   

   Not Hispanic or Latino but Black or African American alone  \
0                                              14461            
1                                               8015            
2                                              11275            
3                                               1488            
4                                                738            

   Not Hispanic or Latino but American Indian and Alaska Native alone  \
0                                                142                    
1                                                189                    
2                                                191                    
3                                                 72                    
4                                                 55                    

   Not Hispanic or Latino but Asian alone  \
0                                    3523   
1                                   16184   
2                                    6414   
3                                    4399   
4                                    1449   

   Not Hispanic or Latino but Native Hawaiian and Other Pacific Islander alone  \
0                                                 39                             
1                                                  3                             
2                                                 10                             
3                                                 11                             
4                                                 16                             

   Not Hispanic or Latino but Some Other Race alone  \
0                                               202   
1                                               225   
2                                               130   
3                                               128   
4                                                51   

   Not Hispanic or Latino but Two or More Races  ...  FeMale: 60 to 61 yrs  \
0                                          1573  ...                   486   
1                                          1919  ...                   794   
2                                          1235  ...                   561   
3                                           917  ...                   322   
4                                           594  ...                   261   

   FeMale: 62 to 64 yrs  FeMale: 65 to 66 yrs  FeMale: 67 to 69 yrs  \
0                   661                   308                   383   
1                  1026                   561                   718   
2                   679                   360                   475   
3                   493                   256                   323   
4                   314                   159                   218   

   FeMale: 70 to 74 yrs  FeMale: 75 to 79 yrs  FeMale: 80 to 84 yrs  \
0                   534                   392                   312   
1                  1056                   799                   713   
2                   763                   620                   524   
3                   436                   340                   275   
4                   319                   257                   181   

   FeMale: 85 yrs and over  Median Age  Total Households  
0                      388   32.404941             24292  
1                      931   35.062344             24755  
2                      690  

In [4]:
# Cleaning up of the datasets - Demographics data
##Cleaning up includes, removing those age groups that eat in a restuarant very less, ages < 15yrs and < 80yrs
##Male Cleaning up includes, merging some columns and dropping after merge 
Chicago_Neigh.drop(['Male:  to 5 yrs','Male: 5 to 9 yrs','Male: 10 to 14 yrs'],axis = 1, inplace = True)
Chicago_Neigh['Male: 15 to 70 yrs'] = Chicago_Neigh['Male: 15 to 17 yrs'] + Chicago_Neigh['Male: 18 and 19 yrs'] + Chicago_Neigh['Male: 20 yrs'] + Chicago_Neigh['Male: 21 yrs'] + Chicago_Neigh['Male: 22 to 24 yrs'] + Chicago_Neigh['Male: 25 to 29 yrs'] + Chicago_Neigh['Male: 30 to 34 yrs'] + Chicago_Neigh['Male: 35 to 39 yrs'] + Chicago_Neigh['Male: 40 to 44 yrs'] + Chicago_Neigh['Male: 45 to 49 yrs'] + Chicago_Neigh['Male: 50 to 54 yrs'] + Chicago_Neigh['Male: 55 to 59 yrs'] + Chicago_Neigh['Male: 60 to 61 yrs']+ Chicago_Neigh['Male: 62 to 64 yrs'] + Chicago_Neigh['Male: 65 to 66 yrs'] + Chicago_Neigh['Male: 67 to 69 yrs']
Chicago_Neigh.drop(['Male: 15 to 17 yrs','Male: 18 and 19 yrs','Male: 20 yrs','Male: 21 yrs','Male: 22 to 24 yrs','Male: 25 to 29 yrs','Male: 30 to 34 yrs','Male: 35 to 39 yrs','Male: 40 to 44 yrs','Male: 45 to 49 yrs','Male: 50 to 54 yrs','Male: 55 to 59 yrs','Male: 70 to 74 yrs','Male: 75 to 79 yrs','Male: 80 to 84 yrs','Male: 60 to 61 yrs','Male: 62 to 64 yrs','Male: 65 to 66 yrs','Male: 67 to 69 yrs','Male: 85 yrs and over'],axis = 1, inplace = True)
##Female Cleaning up includes, merging some columns and dropping after merge 
Chicago_Neigh.drop(['FeMale: Under 5 yrs','FeMale: 5 to 9 yrs','FeMale: 10 to 14 yrs','FeMale: 70 to 74 yrs','FeMale: 75 to 79 yrs','FeMale: 80 to 84 yrs','FeMale: 85 yrs and over'],axis = 1, inplace = True)
Chicago_Neigh['Female: 15 to 70 yrs'] = Chicago_Neigh['FeMale: 15 to 17 yrs']+Chicago_Neigh['FeMale: 18 to 19 yrs']+Chicago_Neigh['FeMale: 20 yrs']+Chicago_Neigh['FeMale: 21 yrs']+Chicago_Neigh['FeMale: 22 to 24 yrs']+Chicago_Neigh['FeMale: 25 to 29 yrs']+Chicago_Neigh['FeMale: 30 to 34 yrs']+Chicago_Neigh['FeMale: 35 to 39 yrs']+Chicago_Neigh['FeMale: 40 to 44 yrs']+Chicago_Neigh['FeMale: 45 to 49 yrs']+Chicago_Neigh['FeMale: 50 to 54 yrs']+Chicago_Neigh['FeMale: 55 to 59 yrs']+Chicago_Neigh['FeMale: 60 to 61 yrs']+Chicago_Neigh['FeMale: 62 to 64 yrs']+Chicago_Neigh['FeMale: 65 to 66 yrs']+Chicago_Neigh['FeMale: 67 to 69 yrs']
Chicago_Neigh.drop(['FeMale: 15 to 17 yrs','FeMale: 18 to 19 yrs','FeMale: 20 yrs','FeMale: 21 yrs','FeMale: 22 to 24 yrs','FeMale: 25 to 29 yrs','FeMale: 30 to 34 yrs','FeMale: 35 to 39 yrs','FeMale: 40 to 44 yrs','FeMale: 45 to 49 yrs','FeMale: 50 to 54 yrs','FeMale: 55 to 59 yrs','FeMale: 60 to 61 yrs','FeMale: 62 to 64 yrs','FeMale: 65 to 66 yrs','FeMale: 67 to 69 yrs'],axis = 1, inplace = True)
## Merging all communities
Chicago_Neigh['Not Hispanic or Latino'] = Chicago_Neigh['Not Hispanic or Latino but White alone']+Chicago_Neigh['Not Hispanic or Latino but Black or African American alone']+Chicago_Neigh['Not Hispanic or Latino but American Indian and Alaska Native alone']+Chicago_Neigh['Not Hispanic or Latino but Asian alone']+Chicago_Neigh['Not Hispanic or Latino but Native Hawaiian and Other Pacific Islander alone']+Chicago_Neigh['Not Hispanic or Latino but Some Other Race alone']+Chicago_Neigh['Not Hispanic or Latino but Two or More Races']
Chicago_Neigh.drop(['GeogKey','Not Hispanic or Latino but White alone','Not Hispanic or Latino but Black or African American alone','Not Hispanic or Latino but American Indian and Alaska Native alone','Not Hispanic or Latino but Asian alone','Not Hispanic or Latino but Native Hawaiian and Other Pacific Islander alone','Not Hispanic or Latino but Some Other Race alone','Not Hispanic or Latino but Two or More Races'],axis = 1, inplace = True)
Chicago_Neigh.rename(columns={'Geog':'COMMUNITY NAME'}, inplace=True)
Chicago_Neigh['COMMUNITY NAME'] =  Chicago_Neigh['COMMUNITY NAME'].str.upper()
Chicago_Neigh['Hispanic or Lation % of Total'] = Chicago_Neigh['Hispanic or Latino']/Chicago_Neigh['Total Population'] * 100
Chicago_Neigh.shape

(77, 9)

In [5]:
# Cleaning up of the datasets - Second Demographics data
Chicago_Neigh_Demo.rename(columns={'COMMUNITY AREA NAME':'COMMUNITY NAME'}, inplace=True)
Chicago_Neigh_Demo.drop(['Community Area Number','PERCENT OF HOUSING CROWDED','PERCENT HOUSEHOLDS BELOW POVERTY','PERCENT AGED 16+ UNEMPLOYED','PERCENT AGED 25+ WITHOUT HIGH SCHOOL DIPLOMA','HARDSHIP INDEX'],axis=1, inplace=True)
Chicago_Neigh_Demo['COMMUNITY NAME'] =  Chicago_Neigh_Demo['COMMUNITY NAME'].str.upper()
Chicago_Neigh_Demo.set_index("COMMUNITY NAME")
Chicago_Neigh_Demo.reset_index()
print(Chicago_Neigh_Demo.dtypes)
Chicago_Neigh_Demo.head()

COMMUNITY NAME                       object
PERCENT AGED UNDER 18 OR OVER 64    float64
PER CAPITA INCOME                     int64
dtype: object


COMMUNITY NAME  PERCENT AGED UNDER 18 OR OVER 64  PER CAPITA INCOME 
0     ALBANY PARK                              32.0               21323
1  ARCHER HEIGHTS                              39.2               16134
2   ARMOUR SQUARE                              38.3               16148
3         ASHBURN                              36.9               23482
4  AUBURN GRESHAM                              41.9               15528

In [6]:
# After careful research online, I couldnt find a geocodes dataset for the communities, however I found a shapefile with an option to export to a CSV. Downloaded the CSV and
# altered the file to bring it to a format I need.
#Import GeoCodes CSV
Chicago_geocode = pd.read_csv('Chicago_Lat_Long.csv')
Chicago_geocode.rename(columns={'Community Name':'COMMUNITY NAME'}, inplace=True)
Chicago_geocode['COMMUNITY NAME'] =  Chicago_geocode['COMMUNITY NAME'].str.upper()


## In this case we are merging both result sets to make it ready for 
Chicago_Neigh_stg = pd.merge(Chicago_Neigh_Demo, Chicago_Neigh, on='COMMUNITY NAME', how='outer')
Chicago_Neigh_Full = pd.merge(Chicago_Neigh_stg, Chicago_geocode, on='COMMUNITY NAME', how='outer')
Chicago_Neigh_Full.set_index('COMMUNITY NAME')
Chicago_Neigh_Full.head(5)

COMMUNITY NAME  PERCENT AGED UNDER 18 OR OVER 64  PER CAPITA INCOME   \
0     ALBANY PARK                              32.0               21323   
1  ARCHER HEIGHTS                              39.2               16134   
2   ARMOUR SQUARE                              38.3               16148   
3         ASHBURN                              36.9               23482   
4  AUBURN GRESHAM                              41.9               15528   

   Total Population  Hispanic or Latino  Median Age  Total Households  \
0           51542.0             25487.0   31.078026           16322.0   
1           13393.0             10182.0   30.507134            3844.0   
2           13391.0               464.0   41.118083            5247.0   
3           41081.0             15132.0   33.483019           12451.0   
4           48743.0               459.0   36.969525           17603.0   

   Male: 15 to 70 yrs  Female: 15 to 70 yrs  Not Hispanic or Latino  \
0             19715.0               18065.0                 26055.0   
1              4734.0                4483.0                  3211.0   
2              4725.0                4902.0                 12927.0   
3             13771.0               14939.0                 25949.0   
4             14402.0               18888.0                 48284.0   

   Hispanic or Lation % of Total   Latitude  Longitude  
0                      49.448993  41.961451 -87.696929  
1                      76.024789  41.807749 -87.722579  
2                       3.465014  41.853035 -87.630782  
3                      36.834546  41.749095 -87.679637  
4                       0.941674  41.752451 -87.652969

### Step 2: Create three Ranking columns

    1) Rank on the Hispanic and latino population counts 

    2) Rank on Income of the community 

    3) Rank on the age groups

In [7]:
Chicago_Neigh_Full['Income_Rank'] = Chicago_Neigh_Full['PER CAPITA INCOME '].rank(ascending = 0) 
Chicago_Neigh_Full['H&L_Pop%_Rank'] = Chicago_Neigh_Full['Hispanic or Lation % of Total'].rank(ascending = 0) 
Chicago_Neigh_Full['Age_Rank'] = Chicago_Neigh_Full['PERCENT AGED UNDER 18 OR OVER 64'].rank(ascending = 0) 

####Creating another column which sums the Income rank and Hispanic and Lation Population % ranks as well

Chicago_Neigh_Full['Income_&_Hispanic_Pop_Combined_Rank'] = Chicago_Neigh_Full['Income_Rank']+Chicago_Neigh_Full['H&L_Pop%_Rank']
Chicago_Neigh_Full.head(5)

COMMUNITY NAME  PERCENT AGED UNDER 18 OR OVER 64  PER CAPITA INCOME   \
0     ALBANY PARK                              32.0               21323   
1  ARCHER HEIGHTS                              39.2               16134   
2   ARMOUR SQUARE                              38.3               16148   
3         ASHBURN                              36.9               23482   
4  AUBURN GRESHAM                              41.9               15528   

   Total Population  Hispanic or Latino  Median Age  Total Households  \
0           51542.0             25487.0   31.078026           16322.0   
1           13393.0             10182.0   30.507134            3844.0   
2           13391.0               464.0   41.118083            5247.0   
3           41081.0             15132.0   33.483019           12451.0   
4           48743.0               459.0   36.969525           17603.0   

   Male: 15 to 70 yrs  Female: 15 to 70 yrs  Not Hispanic or Latino  \
0             19715.0               18065.0                 26055.0   
1              4734.0                4483.0                  3211.0   
2              4725.0                4902.0                 12927.0   
3             13771.0               14939.0                 25949.0   
4             14402.0               18888.0                 48284.0   

   Hispanic or Lation % of Total   Latitude  Longitude  Income_Rank  \
0                      49.448993  41.961451 -87.696929         40.0   
1                      76.024789  41.807749 -87.722579         57.0   
2                       3.465014  41.853035 -87.630782         56.0   
3                      36.834546  41.749095 -87.679637         34.0   
4                       0.941674  41.752451 -87.652969         60.0   

   H&L_Pop%_Rank  Age_Rank  Income_&_Hispanic_Pop_Combined_Rank  
0           18.0      59.0                                 58.0  
1           10.0      31.0                                 67.0  
2           58.0      38.0                                114.0  
3           24.0      45.0                                 58.0  
4           75.0      14.0                                135.0

In [8]:
Chicago_Neigh_Full = Chicago_Neigh_Full.fillna(0)
Chicago_Neigh_Full['Income_Rank'] = Chicago_Neigh_Full['Income_Rank'].astype(int)
Chicago_Neigh_Full['H&L_Pop%_Rank'] = Chicago_Neigh_Full['H&L_Pop%_Rank'].astype(int)
Chicago_Neigh_Full['Age_Rank'] = Chicago_Neigh_Full['Age_Rank'].astype(int)
Chicago_Neigh_Full['Income_&_Hispanic_Pop_Combined_Rank'] = Chicago_Neigh_Full['Income_&_Hispanic_Pop_Combined_Rank'].astype(int)
Chicago_Neigh_Full.dtypes
# Chicago_Neigh_Full.sort_values(['Income_Rank','H&L_Pop%_Rank','Age_Rank'],ascending=True)

COMMUNITY NAME                          object
PERCENT AGED UNDER 18 OR OVER 64       float64
PER CAPITA INCOME                        int64
Total Population                       float64
Hispanic or Latino                     float64
Median Age                             float64
Total Households                       float64
Male: 15 to 70 yrs                     float64
Female: 15 to 70 yrs                   float64
Not Hispanic or Latino                 float64
Hispanic or Lation % of Total          float64
Latitude                               float64
Longitude                              float64
Income_Rank                              int64
H&L_Pop%_Rank                            int64
Age_Rank                                 int64
Income_&_Hispanic_Pop_Combined_Rank      int64
dtype: object

In [9]:
####Performing some clean ups.
### In the demographics file there is a a row with "Community Name" == Chicago, it is not needed so taking it off
indexNames = Chicago_Neigh_Full[(Chicago_Neigh_Full["COMMUNITY NAME"] == "CHICAGO")].index
Chicago_Neigh_Full.drop(indexNames , inplace=True)
Chicago_Neigh_Full.head(5)

COMMUNITY NAME  PERCENT AGED UNDER 18 OR OVER 64  PER CAPITA INCOME   \
0     ALBANY PARK                              32.0               21323   
1  ARCHER HEIGHTS                              39.2               16134   
2   ARMOUR SQUARE                              38.3               16148   
3         ASHBURN                              36.9               23482   
4  AUBURN GRESHAM                              41.9               15528   

   Total Population  Hispanic or Latino  Median Age  Total Households  \
0           51542.0             25487.0   31.078026           16322.0   
1           13393.0             10182.0   30.507134            3844.0   
2           13391.0               464.0   41.118083            5247.0   
3           41081.0             15132.0   33.483019           12451.0   
4           48743.0               459.0   36.969525           17603.0   

   Male: 15 to 70 yrs  Female: 15 to 70 yrs  Not Hispanic or Latino  \
0             19715.0               18065.0                 26055.0   
1              4734.0                4483.0                  3211.0   
2              4725.0                4902.0                 12927.0   
3             13771.0               14939.0                 25949.0   
4             14402.0               18888.0                 48284.0   

   Hispanic or Lation % of Total   Latitude  Longitude  Income_Rank  \
0                      49.448993  41.961451 -87.696929           40   
1                      76.024789  41.807749 -87.722579           57   
2                       3.465014  41.853035 -87.630782           56   
3                      36.834546  41.749095 -87.679637           34   
4                       0.941674  41.752451 -87.652969           60   

   H&L_Pop%_Rank  Age_Rank  Income_&_Hispanic_Pop_Combined_Rank  
0             18        59                                   58  
1             10        31                                   67  
2             58        38                                  114  
3             24        45                                   58  
4             75        14                                  135

In [10]:
## Adding Lat and Long information of all the communities
neighbourhood_latitude = Chicago_Neigh_Full.loc[0, 'Latitude'] # neighborhood latitude value
neighbourhood_longitude = Chicago_Neigh_Full.loc[0, 'Longitude'] # neighborhood longitude value
neighbourhood_name = Chicago_Neigh_Full.loc[0, 'COMMUNITY NAME'] # neighborhood name

print('Latitude and longitude values of {} are {}, {}.'.format(neighbourhood_name, 
                                                               neighbourhood_latitude, 
                                                               neighbourhood_longitude))

Latitude and longitude values of ALBANY PARK are 41.961451000000004, -87.696929.


### Step3 : Using Foursquare API to get the Neighbourhood Mexican restuarants and their price tag

In [11]:
CLIENT_ID = '0MWLXE21XWIGAFGOXET21G53OQXRSGUSCV5DFZWDTA5ESPJ5' # your Foursquare ID
CLIENT_SECRET = 'VPTKIT4VQCH5VDZS0OXN2MJIX2JE0LP1ZRADXFQTUUXE55ZS' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: 0MWLXE21XWIGAFGOXET21G53OQXRSGUSCV5DFZWDTA5ESPJ5
CLIENT_SECRET:VPTKIT4VQCH5VDZS0OXN2MJIX2JE0LP1ZRADXFQTUUXE55ZS


In [12]:
LIMIT = 5
radius = 500
categoryId='4bf58dd8d48988d1c1941735'
url = 'https://api.foursquare.com/v2/venues/search?client_id={}&client_secret={}&ll={},{}&v={}&categoryId={}&radius={}&limit={}'.format(CLIENT_ID, CLIENT_SECRET, neighbourhood_latitude, neighbourhood_longitude, VERSION, categoryId, radius, LIMIT)
print(url)

https://api.foursquare.com/v2/venues/search?client_id=0MWLXE21XWIGAFGOXET21G53OQXRSGUSCV5DFZWDTA5ESPJ5&client_secret=VPTKIT4VQCH5VDZS0OXN2MJIX2JE0LP1ZRADXFQTUUXE55ZS&ll=41.961451000000004,-87.696929&v=20180605&categoryId=4bf58dd8d48988d1c1941735&radius=500&limit=5


In [13]:
!pip install geocoder
!pip install geopy
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values
import geocoder # import geocoder
import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

     |████████████████████████████████| 102kB 2.9MB/s ta 0:00:011
     |████████████████████████████████| 829kB 24.7MB/s eta 0:00:01
     |████████████████████████████████| 81kB 20.0MB/s eta 0:00:01
  Stored in directory: /home/jupyterlab/.cache/pip/wheels/0c/61/d2/d6b7317325828fbb39ee6ad559dbe4664d0896da4721bf379e
Successfully built future
     |████████████████████████████████| 102kB 2.9MB/s ta 0:00:011
  Stored in directory: /home/jupyterlab/.cache/pip/wheels/99/45/d1/14954797e2a976083182c2e7da9b4e924509e59b6e5c661061
Successfully built geographiclib


In [14]:
import folium # map rendering library
latitude = 41.961451
longitude = -87.6298
map_chicago = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, community in zip(Chicago_Neigh_Full['Latitude'], Chicago_Neigh_Full['Longitude'], Chicago_Neigh_Full['COMMUNITY NAME']):
    label = '{}'.format(community)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_chicago)  

print('##### Here is a representation of Communities on a map####')
map_chicago

##### Here is a representation of Communities on a map####


In [15]:
###Get the mexican restuarants 
def getNearbyVenues(names, latitudes, longitudes, radius=500):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&categoryId=4bf58dd8d48988d1c1941735&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        # print(results)
        # return only relevant information for each nearby venue
        venues_list.append([(
            v['venue']['id'],
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'], 
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['id',
                  'COMMUNITY NAME', 
                  'COMMUNITY Latitude', 
                  'COMMUNITY Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [17]:
Community_Mex_Rest1 = getNearbyVenues(names="ALBANY PARK",
                                   latitudes='41.961451',
                                   longitudes='-87.696929'
                                  )
Community_Mex_Rest1

A


KeyError: 'groups'

### List of Mexican restuarants

In [18]:
Community_Mex_Rest = getNearbyVenues(names=Chicago_Neigh_Full['COMMUNITY NAME'],
                                   latitudes=Chicago_Neigh_Full['Latitude'],
                                   longitudes=Chicago_Neigh_Full['Longitude']
                                  )
Community_Mex_Rest

ALBANY PARK
ARCHER HEIGHTS
ARMOUR SQUARE
ASHBURN
AUBURN GRESHAM
AUSTIN
AVALON PARK
AVONDALE
BELMONT CRAGIN
BEVERLY
BRIDGEPORT
BRIGHTON PARK
BURNSIDE
CALUMET HEIGHTS
CHATHAM
CHICAGO LAWN
CLEARING
DOUGLAS
DUNNING
EAST GARFIELD PARK
EAST SIDE
EDGEWATER
EDISON PARK
ENGLEWOOD
FOREST GLEN
FULLER PARK
GAGE PARK
GARFIELD RIDGE
GRAND BOULEVARD
GREATER GRAND CROSSING
HEGEWISCH
HERMOSA
HUMBOLDT PARK
HYDE PARK
IRVING PARK
JEFFERSON PARK
KENWOOD
LAKE VIEW
LINCOLN PARK
LINCOLN SQUARE
LOGAN SQUARE
THE LOOP
LOWER WEST SIDE
MCKINLEY PARK
MONTCLARE
MORGAN PARK
MOUNT GREENWOOD
NEAR NORTH SIDE
NEAR SOUTH SIDE
NEAR WEST SIDE
NEW CITY
NORTH CENTER
NORTH LAWNDALE
NORTH PARK
NORWOOD PARK
OAKLAND
O'HARE
PORTAGE PARK
PULLMAN
RIVERDALE
ROGERS PARK
ROSELAND
SOUTH CHICAGO
SOUTH DEERING
SOUTH LAWNDALE
SOUTH SHORE
UPTOWN
WASHINGTON HEIGHTS
WASHINGTON PARK
WEST ELSDON
WEST ENGLEWOOD
WEST GARFIELD PARK
WEST LAWN
WEST PULLMAN
WEST RIDGE
WEST TOWN
WOODLAWN


id  COMMUNITY NAME  COMMUNITY Latitude  \
0    517c6fe4e4b019c93a3978af     ALBANY PARK           41.961451   
1    4bdc8a3a2a3a0f47cd50b3b6     ALBANY PARK           41.961451   
2    4b48d32df964a520c55826e3  ARCHER HEIGHTS           41.807749   
3    4e98ac36775b6787c26bcf80  ARCHER HEIGHTS           41.807749   
4    548dff00498eecad7325ffb8  ARCHER HEIGHTS           41.807749   
..                        ...             ...                 ...   
136  4e879cdc93adfd051d6d609e        WOODLAWN           41.880820   
137  55b112d0498e1605afe59bd4        WOODLAWN           41.880820   
138  49d8159cf964a520a05d1fe3        WOODLAWN           41.880820   
139  553977b7498eb9779dcb0592        WOODLAWN           41.880820   
140  4ef105fdb8f73be2d09fc3fb        WOODLAWN           41.880820   

     COMMUNITY Longitude                        Venue  Venue Latitude  \
0             -87.696929  Emma's Fruits @ Horner Park       41.959496   
1             -87.696929                     Que Rico       41.959259   
2             -87.722579           Birrieria Zaragoza       41.804456   
3             -87.722579                    El Asador       41.807675   
4             -87.722579     Machetes Big Quesadillas       41.804455   
..                   ...                          ...             ...   
136           -87.620640    Wildberry Pancakes & Cafe       41.884412   
137           -87.620640                      LaBamba       41.881561   
138           -87.620640                 Miller's Pub       41.879806   
139           -87.620640        Goddess and The Baker       41.881305   
140           -87.620640       Taco Burrito on Wabash       41.881774   

     Venue Longitude      Venue Category  
0         -87.698189  Mexican Restaurant  
1         -87.699812  Mexican Restaurant  
2         -87.723563  Mexican Restaurant  
3         -87.726896  Mexican Restaurant  
4         -87.720035  Mexican Restaurant  
..               ...                 ...  
136       -87.623047      Breakfast Spot  
137       -87.626136       Burrito Place  
138       -87.625966                 Pub  
139       -87.626117                Café  
140       -87.626170  Mexican Restaurant  

[141 rows x 8 columns]

### Above step concludes that there are 141 restuarants in Chicago altogether that are tagged "Mexican"

In [ ]:
print(Community_Mex_Rest.groupby('COMMUNITY NAME').count())
Chicago_Neigh_Full['Count of Mexican Restuarants'] = Community_Mex_Rest.groupby(['COMMUNITY NAME'])['id'].transform('count')
Chicago_Neigh_Full

In [30]:
###Get the mexican restuarants 
def getPriceByVenues(id):
    
    venues_price_list=[]
    for id in id:
        print(id)
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/{}?&client_id={}&client_secret={}&v={}'.format(id,CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION)
        print(url)
        # make the GET request
        results = requests.get(url).json()['response']
        print(results)
        # return only relevant information for each nearby venue
        venues_price_list.append([(
            v['venue']['id']
            ,v['venue']['price']['message']
            ) for v in results])
        #print(venues_price_list)
    nearby_venues_list = pd.DataFrame([item for venues_price_list in venues_price_list for item in venues_price_list])
    nearby_venues_list.columns = ['id', 
                  'Price_Type' 
                            ]
    
    return(nearby_venues_list)

In [25]:
#Chicago_venues_price = getPriceByVenues(id=Chicago_venues['id'])
#Chicago_Mex_rest = pd.merge(Chicago_venues, Chicago_venues_price, on='id', how='outer')
Community_Mex_Rest_price_tier = getPriceByVenues(id=Community_Mex_Rest['id']
                                  )
Community_Mex_Rest_price_tier

517c6fe4e4b019c93a3978af
https://api.foursquare.com/v2/venues/517c6fe4e4b019c93a3978af?&client_id=0MWLXE21XWIGAFGOXET21G53OQXRSGUSCV5DFZWDTA5ESPJ5&client_secret=VPTKIT4VQCH5VDZS0OXN2MJIX2JE0LP1ZRADXFQTUUXE55ZS&v=20180605
{'venue': {'id': '517c6fe4e4b019c93a3978af', 'name': "Emma's Fruits @ Horner Park", 'contact': {}, 'location': {'address': 'Horner Park', 'crossStreet': 'California And Cullom', 'lat': 41.959495544433594, 'lng': -87.69818878173828, 'labeledLatLngs': [{'label': 'display', 'lat': 41.959495544433594, 'lng': -87.69818878173828}], 'postalCode': '60625', 'cc': 'US', 'city': 'Chicago', 'state': 'IL', 'country': 'United States', 'formattedAddress': ['Horner Park (California And Cullom)', 'Chicago, IL 60625', 'United States']}, 'canonicalUrl': 'https://foursquare.com/v/emmas-fruits--horner-park/517c6fe4e4b019c93a3978af', 'categories': [{'id': '4bf58dd8d48988d1c1941735', 'name': 'Mexican Restaurant', 'pluralName': 'Mexican Restaurants', 'shortName': 'Mexican', 'icon': {'prefix':

TypeError: string indices must be integers

In [ ]:
#Community_Mex_Rest.to_csv(r'Community_Mex_Rest.csv')
Community_Mex_Rest = pd.read_csv('Community_Mex_Rest.csv')
Community_Mex_Rest.drop(['Unnamed: 0'],axis=1, inplace = True)
Community_Mex_Rest.head()

In [ ]:
Chicago_Neigh_Top_List.sort_values(['Income_&_Hispanic_Pop_Combined_Rank'],ascending=True).inplace=True
Chicago_Neigh_Top_List

In [ ]:
Chicago_Mex_Final = pd.merge(Community_Mex_Rest, Chicago_Neigh_Top_List, on='COMMUNITY NAME', how='outer')
# Chicago_Mex_Final.dtypes
Chicago_Mex_Final.sort_values(by=['Income_&_Hispanic_Pop_Combined_Rank'], ascending=False).inplace=True
Chicago_Mex_Final.drop(['id','Male: 15 to 70 yrs','Female: 15 to 70 yrs','Latitude','Longitude','Median Age','Age_Rank','Total Households'],axis = 1, inplace = True)
Chicago_Mex_Final

In [ ]:
Chicago_Mex_Final['Income_&_Hispanic_Pop_Combined_Rank'].sort_values().inplace=True
Chicago_Mex_Final['Rank'] = Chicago_Mex_Final['Income_&_Hispanic_Pop_Combined_Rank'].rank(ascending = 1) 
Chicago_Mex_Final

In [ ]:
indexNames = Chicago_Mex_Final[(Chicago_Mex_Final["COMMUNITY NAME"] == "CLEARING")].index
Chicago_Mex_Final.drop(indexNames , inplace=True)
indexNames = Chicago_Mex_Final[(Chicago_Mex_Final["COMMUNITY NAME"] == "FOREST GLEN")].index
Chicago_Mex_Final.drop(indexNames , inplace=True)

In [ ]:
latitude = 41.961451
longitude = -87.6298
map_chicago_mex = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, community in zip(Chicago_Mex_Final['COMMUNITY Latitude'], Chicago_Mex_Final['COMMUNITY Longitude'], Chicago_Mex_Final['COMMUNITY NAME']):
    label = '{}'.format(community)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_chicago_mex)  

map_chicago_mex

In [ ]:
for lat, lng, community,income_rank,hispanic_pop_pct in zip(Chicago_Mex_Final['Venue Latitude'], Chicago_Mex_Final['Venue Longitude'], Chicago_Mex_Final['Venue'],Chicago_Mex_Final['Income_Rank'],Chicago_Mex_Final['Hispanic or Lation % of Total']):
    label = '{}'.format(community,income_rank,hispanic_pop_pct)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='green',
        fill=False,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(map_chicago_mex)  

map_chicago_mex

# Final Results:
There are only two communities that best fit the need:

1) West Town

2) Logan Square

### Reasons are:

1) There is good Hispanic and Latino Population

2) There is good income levels that can help start a moderate price range restuarant

3) There is good age range that eats outside